packages/flame/lib/src/camera/behaviors/viewport_aware_bounds_behavior.dart

```diff
@@ -0,0 +1,105 @@
+import 'dart:math';
+
+import 'package:flame/extensions.dart';
+import 'package:flame/src/camera/behaviors/bounded_position_behavior.dart';
+import 'package:flame/src/camera/viewfinder.dart';
+import 'package:flame/src/camera/viewport.dart';
+import 'package:flame/src/components/core/component.dart';
+import 'package:flame/src/components/mixins/parent_is_a.dart';
+import 'package:flame/src/experimental/geometry/shapes/circle.dart';
+import 'package:flame/src/experimental/geometry/shapes/rectangle.dart';
+import 'package:flame/src/experimental/geometry/shapes/rounded_rectangle.dart';
+import 'package:flame/src/experimental/geometry/shapes/shape.dart';
+
+/// This behavior ensures that the none of the viewport can go outside
+/// of the bounds, when it is false only the viewfinder anchor is considered.
+/// Note that it only works with [Rectangle], [RoundedRectangle] and [Circle]
+/// shapes.
```

Comment:
```suggestion
/// This behavior ensures that none of the viewport can go outside
/// of the bounds, when it is false only the viewfinder anchor is considered.
/// Note that it only works with [Rectangle], [RoundedRectangle] and [Circle]
/// shapes.
```
When what is false?
Align to 80

Reply:
Sure !